test/conftest.py

```diff
@@ -795,34 +795,55 @@ async def QueueLen(self, stream):
 
     async def SandboxCreate(self, stream):
         request: api_pb2.SandboxCreateRequest = await stream.recv_message()
-        # Not using asyncio.subprocess here for Python 3.7 compatibility.
-        self.sandbox = subprocess.Popen(
-            request.definition.entrypoint_args,
+        if self.is_shell_cmds(request.definition.entrypoint_args):
+            self.is_shell = True
+        self.sandbox = await asyncio.subprocess.create_subprocess_exec(
+            *request.definition.entrypoint_args,
             stdout=asyncio.subprocess.PIPE,
             stderr=asyncio.subprocess.PIPE,
             stdin=asyncio.subprocess.PIPE,
         )
+
         self.sandbox_defs.append(request.definition)
+
         await stream.send_message(api_pb2.SandboxCreateResponse(sandbox_id="sb-123"))
 
+    def is_shell_cmds(self, cmds):
+        shell_list = ["bash", "sh", "/bin/bash"]
+        if len(cmds) == 1 and cmds[0] in shell_list:
+            return True
+        else:
+            return False
```

Comment:
```suggestion
        try:
            return cmds[0] in ["bash", "sh", "/bin/bash"]
        except KeyError:
            return False
```

Reply:
modified to `return len(cmds) == 1 and cmds[0] in ["bash", "sh", "/bin/bash"]` because we don't want the shell prompt in the case of `.spawn_sandbox("bash", "-c", "..."`